DROD/SettingsScreen.cpp

```diff
@@ -1137,41 +1136,6 @@ void CSettingsScreen::OnSelectChange(
 	SynchOption(dwTagNo);
 }
 
-//************************************************************************************
-void CSettingsScreen::RestorePlayerSettings()
-//Upon cancelling, restore settings to the way they were.
-{
-	//Reload player data from database.  This instance of CDbPlayer won't have
-	//changes made on this screen.
-	CDbPlayer *pPlayer = g_pTheDB->GetCurrentPlayer();
-	UpdateWidgetsFromPlayerData(pPlayer);
-	delete pPlayer;
-
-	COptionButtonWidget *pOptionButton;
-	CSliderWidget *pSliderWidget;
-
-	pSliderWidget = DYN_CAST(CSliderWidget*, CWidget*, GetWidget(TAG_GAMMA));
-	g_pTheDBM->SetGamma(pSliderWidget->GetValue());
-
-	pOptionButton = DYN_CAST(COptionButtonWidget*, CWidget*,
-		GetWidget(TAG_ENABLE_SOUNDEFF));
-	g_pTheSound->EnableSoundEffects(pOptionButton->IsChecked());
-	pSliderWidget = DYN_CAST(CSliderWidget*, CWidget*,	GetWidget(TAG_SOUNDEFF_VOLUME));
-	g_pTheSound->SetSoundEffectsVolume(pSliderWidget->GetValue());
```

Comment:
I don't think we can cull this method entirely.  The active graphic and sound settings should be reverted back to their original values, correct?  Sounds like the trick is to do so without redrawing the updated widgets on-screen.

Reply:
Crap, I checked only a few of the lines randomly and it seemed like they don't save any options and of course that was wrong. Updated with a sensible fix!